src/common.jl

```diff
@@ -40,29 +41,32 @@ function DiffEqBase.solve(
         error("TaylorIntegration is not compatible with callbacks.")
     end
 
-    if typeof(prob.u0) <: Number
-        u0 = [prob.u0]
-    else
-        u0 = vec(deepcopy(prob.u0))
-    end
-
     sizeu = size(prob.u0)
-    p = prob.p
     f = prob.f
 
-    if !isinplace && (typeof(prob.u0)<:Vector{Float64} || typeof(prob.u0)<:Number)
-        f! = (t, u, du) -> (du .= f(u, p, t); 0)
-    elseif !isinplace && typeof(prob.u0)<:AbstractArray
-        f! = (t, u, du) -> (du .= vec(f(reshape(u, sizeu), p, t)); 0)
-    elseif typeof(prob.u0)<:Vector{Float64}
-        f! = (t, u, du) -> f(du, u, p, t)
-    else # Then it's an in-place function on an abstract array
-        f! = (t, u, du) -> (f(reshape(du, sizeu), reshape(u, sizeu), p, t);
-                            u = vec(u); du=vec(du); 0)
-    end
+    if !isinplace && typeof(prob.u0) <: Vector{Float64}
+        f! = (du, u, p, t) -> (du .= f(u, p, t); 0)
+        t, vectimeseries = taylorinteg(f!, prob.u0, prob.tspan[1], prob.tspan[2],
+            alg.order, abstol, prob.p, maxsteps=maxiters)
+
+    elseif !isinplace && typeof(prob.u0) <: AbstractArray
+        f! = (du, u, p, t) -> (du .= vec(f(reshape(u, sizeu), p, t)); 0)
+        t, vectimeseries = taylorinteg(f!, prob.u0, prob.tspan[1], prob.tspan[2],
+            alg.order, abstol, prob.p, maxsteps=maxiters)
 
-    t,vectimeseries = taylorinteg(f!, u0, prob.tspan[1], prob.tspan[2], alg.order,
-                                                      abstol, maxsteps=maxiters)
+    elseif isinplace && typeof(prob.u0) <: AbstractArray{eltype(uType),2}
+        f! = (du, u, p, t) -> (
+                dd = reshape(du, sizeu); uu = reshape(u, sizeu);
+                f(dd, uu, p, t); u .= vec(uu); du .= vec(dd); 0)
```

Comment:
You don't need to `u .= vec(uu); du .= vec(dd)` since a `reshape` is a view.

Reply:
Thanks Chris! I'll change it